src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -15,6 +16,9 @@ namespace Microsoft.Extensions.Configuration
     public static class ConfigurationBinder
     {
         private const BindingFlags DeclaredOnlyLookup = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static | BindingFlags.DeclaredOnly;
+        private const string TrimmingWarningMessage = "In case the type is Generic, the trimmer cannot statically analyze the generic type arguments so their members may be trimmed.";
```

Comment:
If I understand the code correctly it will recursively walk properties, so any type is problematic. It doesn't matter if it's generic or not. It's basically a deserializer - having the same issues with trimming as other deserializers.

The support for collections just makes this a harder problem - since it will recursively walk into collection item types.

So the attributes are correct, but the messages are misleading.

Reply:
Maybe "In case the type is non-primitive, ..." ?